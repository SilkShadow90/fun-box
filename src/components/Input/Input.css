.input {
    height: 46px;
}
