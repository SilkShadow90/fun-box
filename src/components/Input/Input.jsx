import React from 'react';
import TextField from '@material-ui/core/TextField';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
    display: 'flex',
    minHeight: '3rem',
  },
});

const Input = React.memo((props) => {
  const {
    classes,
    label,
    value,
    onChange,
    placeholder,
  } = props;

  return (
    <TextField
      label={label}
      value={value}
      margin="dense"
      inputProps={{
        ...props,
        onChange,
        placeholder,
      }}
      className={classNames(classes.textField)}
    />
  );
});

Input.propTypes = {
  classes: PropTypes.shape({
    textField: PropTypes.string.isRequired,
  }).isRequired,
  placeholder: PropTypes.string,
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  value: PropTypes.string,
};

Input.defaultProps = {
  value: '',
  placeholder: '',
  onChange: undefined,
};

export default withStyles(styles)(Input);
