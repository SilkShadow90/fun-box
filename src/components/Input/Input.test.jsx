import React from 'react';
import Renderer from 'react-test-renderer';

import Input from './Input';

describe('check Input component', () => {
  const renderer = Renderer.create(
    <Input
      placeholder="test"
      label="test label"
    />,
  );

  const input = renderer.toJSON();

  test('check expect match type', () => {
    expect(input.type).toBe('div');
  });

  test('check expect match children', () => {
    expect(input.children[0].type).toBe('label');
    expect(input.children[0].children[0]).toBe('test label');
    expect(input.children[1].type).toBe('div');
    expect(input.children[1].children[0].type).toBe('input');
    expect(input.children[1].children[0].props.placeholder).toBe('test');
  });

  test('check expect match snapshot', () => {
    expect(input).toMatchSnapshot();
  });
});
