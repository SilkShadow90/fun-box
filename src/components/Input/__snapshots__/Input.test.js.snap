// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check Input component check expect match children 1`] = `
<div
  className="MuiFormControl-root-2 MuiFormControl-marginDense-4 Input-textField-1"
>
  <label
    className="MuiFormLabel-root-17 MuiInputLabel-root-6 MuiInputLabel-formControl-11 MuiInputLabel-animated-14 MuiInputLabel-marginDense-12"
    data-shrink={false}
  >
    test label
  </label>
  <div
    className="MuiInputBase-root-37 MuiInput-root-24 MuiInput-underline-28 MuiInputBase-formControl-38 MuiInput-formControl-25 MuiInputBase-marginDense-44"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input-47 MuiInput-input-32 MuiInputBase-inputMarginDense-48 MuiInput-inputMarginDense-33"
      classes={
        Object {
          "textField": "Input-textField-1",
        }
      }
      disabled={false}
      label="test label"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="test"
      required={false}
      type="text"
    />
  </div>
</div>
`;

exports[`check Input component check expect match snapshot 1`] = `
<div
  className="MuiFormControl-root-2 MuiFormControl-marginDense-4 Input-textField-1"
>
  <label
    className="MuiFormLabel-root-17 MuiInputLabel-root-6 MuiInputLabel-formControl-11 MuiInputLabel-animated-14 MuiInputLabel-marginDense-12"
    data-shrink={false}
  >
    test label
  </label>
  <div
    className="MuiInputBase-root-37 MuiInput-root-24 MuiInput-underline-28 MuiInputBase-formControl-38 MuiInput-formControl-25 MuiInputBase-marginDense-44"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input-47 MuiInput-input-32 MuiInputBase-inputMarginDense-48 MuiInput-inputMarginDense-33"
      classes={
        Object {
          "textField": "Input-textField-1",
        }
      }
      disabled={false}
      label="test label"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="test"
      required={false}
      type="text"
    />
  </div>
</div>
`;
